src/v/cluster/tests/randoms.h

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/health_monitor_types.h"
+#include "model/tests/randoms.h"
+#include "random/generators.h"
+#include "storage/tests/randoms.h"
+#include "test_utils/randoms.h"
+
+namespace cluster::node {
+
+inline local_state random_local_state() {
+    return local_state{
+      {},
+      tests::random_named_string<application_version>(),
+      tests::random_named_int<cluster::cluster_version>(),
+      tests::random_duration<std::chrono::milliseconds>(),
+      tests::random_vector(storage::random_disk),
+      storage::disk_space_alert{
+        0} // TODO: replace with storage::random_disk_space_alert()
```

Comment:
If storage::disk_space_alert != 0 then compat's self test fails. Anyone know why?

Reply:
yeh i think it is ok for now. the adl encodings are a bit weird so this makes all the stars align.